src/wallet/test/fuzz/wallet_bdb_parser.cpp

```diff
@@ -0,0 +1,108 @@
+#include <test/util/setup_common.h>
+#include <util/fs.h>
+#include <util/time.h>
+#include <util/translation.h>
+#include <wallet/bdb.h>
+#include <wallet/db.h>
+#include <wallet/dump.h>
+#include <wallet/migrate.h>
+
+#include <fstream>
+#include <iostream>
+
+using wallet::DatabaseOptions;
+using wallet::DatabaseStatus;
+
+namespace {
+TestingSetup* g_setup;
+} // namespace
+
+void initialize_wallet_bdb_parser()
+{
+    static auto testing_setup = MakeNoLogFileContext<TestingSetup>();
+    g_setup = testing_setup.get();
+}
+
+FUZZ_TARGET(wallet_bdb_parser, .init = initialize_wallet_bdb_parser)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+
+    const auto wallet_path = g_setup->m_args.GetDataDirNet() / "fuzzed_wallet.dat";
+
+    {
+        AutoFile outfile{fsbridge::fopen(wallet_path, "wb")};
+        const auto file_data{ConsumeRandomLengthByteVector(fuzzed_data_provider)};
+        outfile << Span{file_data};
+    }
+
+    const DatabaseOptions options{};
+    DatabaseStatus status;
+    bilingual_str error;
+
+    fs::path bdb_ro_dumpfile{g_setup->m_args.GetDataDirNet() / "fuzzed_dumpfile_bdb_ro.dat"};
+    if (fs::exists(bdb_ro_dumpfile)) { // Writing into an existing dump file will throw an exception
+        remove(bdb_ro_dumpfile);
+    }
+    g_setup->m_args.ForceSetArg("-dumpfile", fs::PathToString(bdb_ro_dumpfile));
+
+    bool bdb_ro_err = false;
+    try {
+        auto db{MakeBerkeleyRODatabase(wallet_path, options, status, error)};
+        assert(DumpWallet(g_setup->m_args, *db, error));
+    }
+    catch (const std::runtime_error& e) {
+        bdb_ro_err = true;
+        if (std::string(e.what()) == "AutoFile::ignore: end of file: iostream error") return;
+        if (std::string(e.what()) == "AutoFile::read: end of file: iostream error") return;
+        if (std::string(e.what()) == "Not a BDB file") return;
+        if (std::string(e.what()) == "Unsupported BDB data file version number") return;
+        if (std::string(e.what()) == "Unexpected page type, should be 9 (BTree Metadata)") return;
+        if (std::string(e.what()) == "Unexpected database flags, should only be 0x20 (subdatabases)") return;
+        if (std::string(e.what()) == "Unexpected outer database root page type") return;
+        if (std::string(e.what()) == "Unexpected number of entries in outer database root page") return;
+        if (std::string(e.what()) == "Subdatabase has an unexpected name") return;
+        if (std::string(e.what()) == "Subdatabase page number has unexpected length") return;
+        if (std::string(e.what()) == "Unexpected inner database page type") return;
+        if (std::string(e.what()) == "Unknown record type in records page") return;
+        if (std::string(e.what()) == "Unknown record type in internal page") return;
+        if (std::string(e.what()) == "Unexpected page size") return;
+        if (std::string(e.what()) == "Unexpected page type") return;
+        throw e;
+    }
```

Comment:
So far my corpus doesn't make it here, as tested by sprinkling `assert(false)`. I may indeed need some help with a real wallet.